### PR TITLE
vsrpc: Expose spinner updates in VS

### LINKED ISSUE #3313: [AspireDeployment] EnvironmentSevice.DeployAsync() ProgressMessage should correctly set MessageKind property
The `ProgressMessage` sent back from `EnvironmentService.DeployAsync()` should correctly set the `Kind(MessageKind)` property,

From VS IDE, the `Kind` property of the `ProgressMessage` is used to handle the message according to the value (`Logging (0)` or `Important (1)`). 

This is, if the message is `Important`, the information would be shown _prominently_, other than just print it out in the azd `Output` window.

### PR DESCRIPTION
**Pull request by weikanglim**
Expose spinner updates to VS.

Progress messages that are marked as Important are prominently displayed in VS as a spinner:
image

While doing this, `lineWriter` is updated to support line-ending trimming and reduced memory allocations by reusing pre-allocated buffers.

Fixes #3313

### CODE REVIEW

File: cli/azd/internal/vsrpc/message_writer.go [hunk 1]
```diff
@@ -4,21 +4,37 @@
...
 	"sync"
 )
 
 // messageWriter is an io.Writer that writes to an IObserver[ProgressMessage], emitting a message for each write.
 type messageWriter struct {
-	ctx      context.Context
-	observer IObserver[ProgressMessage]
+	ctx             context.Context
+	observer        IObserver[ProgressMessage]
+	messageTemplate ProgressMessage
```
Comment (ellismg) [inline cli/azd/internal/vsrpc/message_writer.go:17]: This + `.Fill` is really clever.

File: cli/azd/internal/vsrpc/models.go [hunk 2]
```diff
@@ -68,6 +68,12 @@ func newInfoProgressMessage(message string) ProgressMessage {
 	}
 }
 
+func (m ProgressMessage) Fill(message string) ProgressMessage {
```
Comment (ellismg) [inline cli/azd/internal/vsrpc/models.go:71]: FWIW - Took me a moment to realize that `Fill` would not actually mutate this instance and instead returns a new instance with the given message and Time.Now().  While the signature makes it clear, to me `Fill` evokes the idea that we would mutate this instance.  I'm not if `CopyWithMessage` is a "better" name, maybe just a helpful doc comment about how this returns a new instance?

**Review by ellismg — APPROVED** [review]
This is a really surgical and nice solution to the problem - I had imagined we would have to do something much more pervasive.  It is nice to see how far we are able to get with the `io.Writer` interface.

I felt like a I learned a lot by reading this change.

File: cli/azd/internal/vsrpc/models.go (hunk 2 above)
Comment (weikanglim) [reply cli/azd/internal/vsrpc/models.go]: Updated to `WithMessage`

File: cli/azd/internal/vsrpc/message_writer.go (hunk 1 above)
Comment (weikanglim) [reply cli/azd/internal/vsrpc/message_writer.go:17]: Thanks for the kind words!

**Pull request merged**